pyresample/future/spherical/extent.py

```diff
@@ -0,0 +1,300 @@
+        raise ValueError("'extent' must have length 4: [lon_min, lon_max, lat_min, lat_max].")
+    # Convert to np.array
+    extent = np.array(extent)
+    # Check order and values
+    _check_extent_order(extent)
+    _check_extent_values(extent, use_radians=use_radians)
+    # Check is not point or line
+    _check_is_not_point_extent(extent)
+    _check_is_not_line_extent(extent)
+    # Ensure is a list
+    extent = extent.tolist()
+    return extent
+
+
+def _check_non_overlapping_extents(polygons):
+    """Check that the extents polygons are non-overlapping.
+
+    Note: Touching extents are considered valids.
+    """
+    for poly in polygons:
+        # Intersects includes within/contain/equals/touches !
+        n_intersects = np.sum([p.intersects(poly) and not p.touches(poly) for p in polygons])
+        if n_intersects >= 2:
+            raise ValueError("The extents composing SExtent can not be duplicates or overlapping.")
+
+
+def _is_global_extent(polygons):
+    """Check if a list of extents composes a global extent."""
+    from shapely.geometry import Polygon
+    from shapely.ops import unary_union
+
+    # Try union the polygons
+    unioned_polygon = unary_union(polygons)
+    # If still a MultiPolygon, not a global extent
+    if not isinstance(unioned_polygon, Polygon):
+        return False
+    # Define global polygon
+    global_polygon = Polygon.from_bounds(-180, -90, 180, 90)
+    # Assess if global
+    is_global = global_polygon.equals(unioned_polygon)
+    return is_global
+
+
+class SExtent(object):
+    """Spherical Extent.
+
+    SExtent longitudes are defined between -180 and 180 degree.
+    Geometrical operations are performed on a planar coordinate system.
+
+    A spherical geometry crossing the anti-meridian will have an SExtent
+     composed of [lon_start, 180, ...,...] and [-180, lon_end, ..., ...]
+
+    Important notes:
+    - SExtents touching at the anti-meridian are considered to not touching !
+    - SExtents.intersects predicate includes also contains/within cases.
+    - In comparison to shapely polygon behaviour, SExtent.intersects is
+      False if SExtents are just touching.
+
+    There is not an upper limit on the number of extents composing SExtent !!!
+    The only conditions is that the extents do not intersect/overlap each other.
+
+    More specifically, extents composing an SExtent:
+    - must represent a polygon (not a point or a line),
+    - can touch each other,
+    - but can not intersect/contain/overlap each other.
+
+    Examples of valid SExtent definitions:
+      extent = [x_min, x_max, y_min, ymax]
+      sext = SExtent(extent)
+      sext = SExtent([extent, extent])
+      sext = SExtent(extent, extent, ...)
+
+    """
+
+    def __init__(self, *args):
+        # Get list of extents
+        list_extents = _get_list_extents_from_args(args)
+        # Check extents format and value validity
+        self.list = [_check_valid_extent(ext) for ext in list_extents]
```

Comment:
Please use a name that isn't a keyword in the Python language/standard library. And does it need to be public (can it be prefixed with `_`)?